### PR TITLE
allow os-independent development and execution, fix silent logging

### PR DESCRIPTION
**Pull request by piax93**
Having `python-apt` as a dependency can result in this tool being tricky to get running in most places (sometimes even on Ubuntu systems). Since the package was only employed to compare Debian package version strings, I think it could a reasonable thing to have a pure Python implementation doing it, and `pydpkg` seems to be able to provide just that.

In doing this, I also fixed a couple other things in the tests:
* Made them independent from os-specific modules, i.e. `lsb_release`.
* Fixed "silent logging" test case as `.hasHandlers()` checks for handlers in all the logging hierarchy (same goes for most methods in the `logging` module), so if the root logger was already set up the `NullHandler` would not be added. Setting the `propagate` property to `False` will isolate the child logger from its parent and avoid that behaviour.

### CODE REVIEW

**Comment by mssalvatore** [comment]
@piax93 Thanks for the contribution. Could you split this into 3 commits?

1) Replace `python-apt` with `pydpkg`
2) Fix silent logging test
3) make test independent from `lsb_release`

**Comment by piax93** [comment]
Ack, I'll do the splitting

**Comment by piax93** [comment]
Rebased changes on `v3.0.0` branch

**Comment by piax93** [comment]
@mssalvatore Any more feedback on this?

**Comment by mssalvatore** [comment]
> @mssalvatore Any more feedback on this?
@piax93, just the 2 code review comments in setup.py.

File: setup.py [hunk 1]
```diff
@@ -29,19 +36,12 @@
         "Topic :: Security",
     ],
     install_requires=[
+        "pydpkg",
```
Comment (mssalvatore) [inline setup.py:39]: Please move this change to https://github.com/canonical/sec-cvescan/pull/50/commits/1d15d9699a90a64ceb765062c83b69ec37467e6f

**Comment by piax93** [comment]
@mssalvatore https://github.com/canonical/sec-cvescan/commit/f1704d849ea948719993d73ff71568bab986ec6d solved an issue I was having with a class being imported in the module's `__init__.py`, issue which seem to be now solved by https://github.com/canonical/sec-cvescan/commit/469103f236f4a662f5ad16c80b4edf55bab9f112.
Anyway, it is a best practice not to import the package itself during the install process because it makes it easy to fall into the trap of indirectly referencing a downstream dependency (https://packaging.python.org/guides/single-sourcing-package-version/, https://stackoverflow.com/a/7071358). Many like to just `exec` the version file, but I personally prefer a simple regex as I think it's cleaner.
I honestly do not have anything against removing that commit, but it does not do any harm either.

File: setup.py (hunk 1 above)
Comment (piax93) [reply setup.py:39]: 🤦 Totally missed this bit, fixing

**Comment by mssalvatore** [comment]
Ideally, setup.py be ignorant of the inner workings of version.py. For example, the regex you used seems like the obvious solution, but it excludes PEP 440 compliant versions like `3.0.0beta1`. If everything just calls get_version(), any bugs can be fixed in 1 place.

If the import in setup.py isn't actually causing any issues ATM, I'd prefer to leave it as-is. Since version.py has a very specific purpose, it's unlikely in the foreseeable future that it will import any downstream dependencies.

**Comment by piax93** [comment]
@mssalvatore My bad on the regex, I alway use numeric only version numbers in my projects and the muscle memory made me fail to be PEP compliant. Anyhow, removed that commit from the branch.

**Comment by mssalvatore** [comment]
There's a dependency version conflict as validators requires six >= 1.14.0 but pydpkg requires exactly six == 1.10.0. I'm waiting on a response to this issue: https://github.com/TheClimateCorporation/python-dpkg/issues/13

**Comment by piax93** [comment]
👍 I'll try to facilitate that

**Pull request merged**